core/src/main/scala/org/apache/spark/sql/execution/datasources/StoreDataSourceStrategy.scala

```diff
@@ -42,25 +42,10 @@ private[sql] object StoreDataSourceStrategy extends Strategy{
         filters,
         t.numPartitions,
         t.partitionColumns,
-        (a, f) => toCatalystRDD(l, a, t.buildScan(a.map(_.name).toArray, f))) :: Nil
-
+        (a, f) => t.buildUnsafeScan(a.map(_.name).toArray, f)) :: Nil
```

Comment:
Are you doing it with your patch or do you want me to do it? Either way is fine with me.